test/utils/test_get_openai_tool_schema.py

```diff
@@ -0,0 +1,230 @@
+                            default_enum_para: RoleType = RoleType.CRITIC,
+                            **kwargs):
+        """
+        A function to test all parameter type.
+        The parameters will be provided by user.
+        Args:
+            str_para (str) : str_para desc
+            int_para (int): int_para desc
+            list_para (List): list_para desc
+            float_para (float): float_para desc
+            datatime_para (datetime): datatime_para desc
+            default_enum_para (RoleType): default_enum_para desc
+        """
+
+    expect_res = {
+        'type': 'function',
+        'function': {
+            'name': 'test_all_parameters',
+            'description': 'A function to test all parameter type.'
+            'The parameters will be provided by user.',
+            'parameters': {
+                '$defs': {
+                    'RoleType': {
+                        'enum': [
+                            'assistant', 'user', 'critic', 'embodiment',
+                            'default'
+                        ],
+                        'type':
+                        'string'
+                    }
+                },
+                'properties': {
+                    'str_para': {
+                        'type': 'string',
+                        'description': 'str_para desc'
+                    },
+                    'int_para': {
+                        'type': 'integer',
+                        'description': 'int_para desc'
+                    },
+                    'list_para': {
+                        'items': {
+                            'type': 'integer'
+                        },
+                        'type': 'array',
+                        'description': 'list_para desc'
+                    },
+                    'float_para': {
+                        'type': 'number',
+                        'description': 'float_para desc'
+                    },
+                    'datatime_para': {
+                        'format': 'date-time',
+                        'type': 'string',
+                        'description': 'datatime_para desc'
+                    },
+                    'default_enum_para': {
+                        'allOf': [{
+                            '$ref': '#/$defs/RoleType'
+                        }],
+                        'default': 'critic',
+                        'description': 'default_enum_para desc'
+                    }
+                },
+                'required': [
+                    'str_para', 'int_para', 'list_para', 'float_para',
+                    'datatime_para'
+                ],
+                'type':
+                'object'
+            }
+        }
+    }
+
+    openai_tool_schema = get_openai_tool_schema(test_all_parameters)
+
+    assert openai_tool_schema == expect_res
+
+    client = OpenAI()
+    expect_res = json.loads("""{
```

Comment:
It may not necessary to test with the backend model here.

Reply:
Initially, my purpose in conducting this test was to verify whether the enumeration type could be correctly accepted by the OpenAI function call. Of course, I think it can be omitted to streamline the test.